File: src/App/App.Api/Services/v1/ClientAuthService.cs
using Dyvenix.App1.App.Api.Contracts.v1;
using Dyvenix.App1.Auth.Shared.ApiClients.v1;
using Dyvenix.App1.Auth.Shared.DTOs;
using Dyvenix.App1.Auth.Shared.Requests.v1;
using Microsoft.Extensions.Logging;

namespace Dyvenix.App1.App.Api.Services.v1;

public class ClientAuthService(TenantApiClient tenantApiClient, ILogger<ClientAuthService> logger) : IClientAuthService
{
	public async Task<TenantDto?> GetTenantById(Guid id)
	{
		return await tenantApiClient.GetById(id);
	}

	public async Task<TenantDto?> GetTenantBySlug(string slug)
	{
		return await tenantApiClient.GetBySlug(slug);
	}

	public async Task<IReadOnlyList<TenantDto>> GetAllTenants()
	{
		return await tenantApiClient.GetAll();
	}

	public async Task<Guid> CreateTenant(CreateTenantReq request)
	{
		return await tenantApiClient.Create(request);
	}

	public async Task UpdateTenant(UpdateTenantReq request)
	{
		await tenantApiClient.Update(request);
	}

	public async Task DeleteTenant(Guid id)
	{
		await tenantApiClient.Delete(id);
	}
}

File: src/UI/Angular/src/app/core/services/app/client-auth.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { environment } from 'environments/environment';
import { Result } from '../common/dtos';
import { unwrapResult, unwrapVoidResult } from '../common/unwrap-result';
import { TenantDto } from './dto';
import { CreateTenantReq, UpdateTenantReq } from './req';

@Injectable({ providedIn: 'root' })
export class ClientAuthService {
	private _httpClient = inject(HttpClient);
	private readonly _baseUrl = `${environment.apiBaseUrl}/api/app/v1/clientauth`;

	// Read - Single

	getTenantById(id: string): Promise<TenantDto | null> {
		return unwrapResult(this._httpClient.get<Result<TenantDto | null>>(`${this._baseUrl}/GetTenantById/${id}`));
	}

	getTenantBySlug(slug: string): Promise<TenantDto | null> {
		return unwrapResult(this._httpClient.get<Result<TenantDto | null>>(`${this._baseUrl}/GetTenantBySlug/${encodeURIComponent(slug)}`));
	}

	// Read - List

	getAllTenants(): Promise<TenantDto[]> {
		return unwrapResult(this._httpClient.get<Result<TenantDto[]>>(`${this._baseUrl}/GetAllTenants`));
	}

	// Create/Update

	createTenant(request: CreateTenantReq): Promise<string> {
		return unwrapResult(this._httpClient.post<Result<string>>(`${this._baseUrl}/CreateTenant`, request));
	}

	updateTenant(request: UpdateTenantReq): Promise<void> {
		return unwrapVoidResult(this._httpClient.put<Result<void>>(`${this._baseUrl}/UpdateTenant`, request));
	}

	// Delete

	deleteTenant(id: string): Promise<void> {
		return unwrapVoidResult(this._httpClient.delete<Result<void>>(`${this._baseUrl}/DeleteTenant`, { body: { id } }));
	}
}
